api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -77,4 +79,26 @@ public boolean isUsernameTaken(String username) {
       throw new RuntimeException(e);
     }
   }
+
+  public void createUser(String givenName, String familyName, String username, String password) {
+    String gSuiteDomain = configProvider.get().googleDirectoryService.gSuiteDomain;
+    User user = new User()
+      .setPrimaryEmail(username+"@"+gSuiteDomain)
+      .setPassword(password)
+      .setName(new UserName().setGivenName(givenName).setFamilyName(familyName));
+    try {
+      getGoogleDirectoryService().users().insert(user).execute();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
So I'm not sure how it comes back, but there are a few errors we're potentially going to want to catch somewhere and handle specially:

- Username already taken
- Invalid username
- Invalid password

If you don't want to mess with that now, let's add a TODO.

Reply:
Good call. I added a TODO locally and might fix it before I call this done.

Reply:
Added a bug to our backlog to handle errors correctly.